src/message/fetchActions.js

```diff
@@ -222,6 +227,42 @@ const fetchTopMostNarrow = () => async (dispatch: Dispatch, getState: GetState)
   }
 };
 
+/**
+ * Calls an async function and if unsuccessful retries the call.
+ *
+ * If the function is an API call and the response has HTTP status code 4xx
+ * the error is considered unrecoverable and the exception is rethrown, to be
+ * handled further up in the call stack.
+ */
+export async function tryFetch<T>(func: () => Promise<T>): Promise<T> {
```

Comment:
The jsdoc should grow a mention of the timeout.

Reply:
Ah, right; thanks, I'll do that.